eclipse-che/che-theia
# fix(plugins-view): Register custom plugin registries from devfile plugins

Signed-off-by: Igor Vinokur <ivinokur@redhat.com>



### What does this PR do?
Scan devfile plugins and if it has custom plugin registry source, add it's plugin registry to che-theia's plugin registry cache.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19898

### How to test this PR?

1. Start a workspace from this devfile:
```
apiVersion: 1.0.0
metadata:
  name: wksp-uhias
components:
  - type: cheEditor
    reference: 'https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-1131/che_theia_meta.yaml'
    alias: che-theia
  - id: editorconfig/editorconfig/0.14.4
    type: chePlugin
    registryUrl: 'http://vinokur-registry.surge.sh/path/to/registry/v3/'
```
2. Open the Plugins view `View => Plugins`.
3. Click the three dots button and select `Show installed plugins`.
See: the plugin `editorconfig` is shown in the list, the icon is displayed as well.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
